agent/structs/discovery_chain.go

```diff
@@ -212,7 +212,7 @@ func (t DiscoveryTarget) String() string {
 	if t.Namespace != "" {
 		b.WriteString(t.Namespace)
 	} else {
-		b.WriteString("default")
+		b.WriteString("<default>")
```

Comment:
Why the `<>`? We treat "default" as the full name for the implicit default namespace in general and just print it in places where it's needed e.g. SPIFFE URIs: `spiffe://<uuid>.consul/ns/default/dc/dc1/svc/web`

Reply:
This stringing method is only used incidentally in tests and nowhere else. I wanted to disambiguate between `""` and `"default"` when displaying data because this is a value type and the real values of all of the struct fields are being compared.

Plumbing wise I _thought_ I was ensuring the `Namespace` field was never empty, so that I wouldn't end up in a situation like: `Target{Service:"a"} != Target{Service:"a", Namespace:"default"}` was happening.

The bug I was chasing ended up being elsewhere, but I wanted to make that sort of thing easier to diagnose.